### PR TITLE
Disable slimefun in world that contains illegal charcters.

### LINKED ISSUE #3874: Slimefun_Not_Working_As_Intended
### ❗ Checklist

- [X] I am using the official english version of Slimefun and did not modify the jar.
- [X] I am using an up to date "DEV" (not "RC") version of Slimefun.
- [X] I am aware that issues related to Slimefun addons need to be reported on their bug trackers and not here.
- [X] I searched for similar open issues and could not find an existing bug report on this.

### 📍 Description

When a member logged on and messed with the Slimefun components, one output chest was working while the other was not and when he tried to use an energy regulator it did not show him as owner rather it shown CS-CoreLib as owner and when he tried to destroy it and place it back it just shown it as CS-CoreLib's head rather than an actual component including the "Connecting..." hologram as well as the Simple Elevator when trying to be used it doesn't work and when destroyed and put into inventory it says Block of Quartz. After these problems rose, I saw in the console a message that states "[20:15:08 ERROR]: [Slimefun] An Exception was caught while ticking the Block Tickers Task for Slimefun vDEV - 1068 (git cab97b1a)
java.lang.IllegalStateException: World "Teknocraft S3.5" seems to have been deleted. Do not call unsafe methods directly!" Even though in the world settings folder of Slimefun has the world on it. The software used is a mixture of different softwares combined: Spigot, Paper, Purpur and Bukkit.

### 📑 Reproduction Steps

1. Updated  to the latest version
2. Uploaded it to my server's file manager
3. Restarted the server
4. Logged on and the issues still occured

### 💡 Expected Behavior

Was expecting for everything to work as normal after these issues, but alas, they still remain.

### 📷 Screenshots / Videos

SimpleElevatorIssue
WorldSettingsSlimefun
CoreLib'sHead
AnExceptionWasCaught-Slimefun
Owner_CSCoreLib

### 📜 Server Log

_No response_

### 📂 `/error-reports/` folder

_No response_

### 💻 Server Software

Other (please specify in your description)

### 🎮 Minecraft Version

1.19.x

### ⭐ Slimefun version

Slimefun4 - DEV 1068 (git cab97)

### 🧭 Other plugins

SimpleUtilities

### PR DESCRIPTION
**Pull request by J3fftw1**
## Description

This PR disabled Slimefun when there is an illegal charcter in 1 of the worlds name.

## Proposed changes

Disable Slimefun when there is an illegal char found

## Related Issues (if applicable)

Resolves #3874 

## Checklist

- [ ] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [ ] I have made sure that the proposed changes do not break compatibility across the supported Minecraft versions (1.16.* - 1.20.*).
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

**Comment by WalshyDev** [comment]
Not sure disabling is the right approach here. Why don't we just not load storage?

Also for the regex I'd probably just do like `[^a-z0-9_-]` so we have an allowlist rather than a blocklist

**Comment by J3fftw1** [comment]
> Not sure disabling is the right approach here. Why don't we just not load storage?
> 
> Also for the regex I'd probably just do like `[^a-z0-9_-]` so we have an allowlist rather than a blocklist

Agreed with the allow list will change that later today.

However for the disabling if we do just not load storage. There will still be a storage folder created and I don’t think we prevent block placing in the worlds with not loaded storage

**Comment by J3fftw1** [comment]
After a little more thought I don’t think an allow list is a good idea. We also support different languages which might support weird characters that aren’t in the English alphabet.
Chinese
Norwegian
German
Etc

**Comment by Sefiraat** [comment]
I do agree that an approach in which we just stop slimefun storing data for an ‘illegal’ world would be better here. 
Assuming that we correctly identify that a world shouldn’t/cant have blockstorage and then do not create a storage object for said world, it should be easy enough in our listeners to stop block placement if there is a valid slimefun item in hand but with no blockstorage object available?

**Review by WalshyDev — COMMENTED** [review]
Not a big fan of this collection approach

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/Slimefun.java [hunk 1]
```diff
@@ -273,6 +280,17 @@ private void onPluginStart() {
...
+        Pattern pattern = Pattern.compile("[\\.\\\\\\/\":><|?*]");
+
+        // Check if world contains an illegal character
+        for (World world : Bukkit.getServer().getWorlds()) {
+            if (pattern.matcher(world.getName()).find()) {
+                illegalWorld.add(world.getName());
+            return;
+            }
+        }
+
```
Comment (JustAHuman-xD) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/Slimefun.java:293]: Is that `return` indented properly?

Comment (WalshyDev) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/Slimefun.java:293]: why are we even returning? That would stop this after the first match

File: src/main/java/io/github/thebusybiscuit/slimefun4/implementation/Slimefun.java [hunk 2]
```diff
@@ -157,6 +159,11 @@ public final class Slimefun extends JavaPlugin implements SlimefunAddon {
      */
     private boolean isNewlyInstalled = false;
 
+    /**
+     * Keep track of if the server has a world file with an illegal character.
+     */
+    private Set<String> illegalWorld = new HashSet<>();
```
Comment (WalshyDev) [inline src/main/java/io/github/thebusybiscuit/slimefun4/implementation/Slimefun.java:165]: how about naming it like disabledWorlds?

Comment (WalshyDev) [reply src/main/java/io/github/thebusybiscuit/slimefun4/implementation/Slimefun.java:165]: this also needs to be final

**Pull request closed without merging**